lib/caching-precompiler.js

```diff
@@ -47,8 +48,29 @@ CachingPrecompiler.prototype._init = function () {
 CachingPrecompiler.prototype._transform = function (code, filePath, hash) {
 	code = code.toString();
 
-	var options = babelConfigHelper.build(this.babelConfig, this.powerAssert, filePath, code);
-	var result = this.babel.transform(code, options);
+	var fileExt = path.extname(filePath).replace(/^\./, '');
+	var isCustomExt = this.extensions.filter(function (ext) {
+		return (ext !== 'js' && ext === fileExt);
+	}).length > 0;
+
+	var result;
+
+	if (isCustomExt) {
+		switch (fileExt) {
+			case 'ts':
+				var extTs = require('./ext/ts');
+				result = extTs.transpile(code, filePath);
+				break;
+
+			default:
+				throw new Error('`--extension ' + fileExt + '` is not supported by AVA.');
+		}
+	}
+
+	if (!result) {
```

Comment:
Should just be an `else`, assuming the body for `if (isCustomExt)` assigns `result` or throws.